### PR TITLE
Remove ERC223 support

### PR DESCRIPTION
**Pull request by hav-noms**

### CODE REVIEW

File: SIPS/sip-30.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+Deprecate [ERC223](https://github.com/ethereum/EIPs/issues/223) from SNX and all Synths to save gas on dex exchanges and no transaction errors.
+
+## Motivation
+
+<!--The motivation is critical for SIPs that want to change Synthetix. It should clearly explain why the existing protocol specification is inadequate to address the problem that the SIP solves. SIP submissions without sufficient motivation may be rejected outright.-->
+
+The UX for [Mintr](https://mintr.synthetix.io) drove the implementation of ERC223 to reduce the number of transactions a user(minter) had to execute to Deposit their sUSD into the Depot FIFO queue to be sold for ETH from 2 to 1 by only eliminating the ERC20 approve transaction prior to calling a ERC20 transferFrom. While this has been a nice UX for mintr users with the [Depot](https://contracts.synthetix.io/Depot)
+
+The benefits of ERC223 transfer have not outweighted the cons on contract to contract transfers;
```
[the remark this answers is not in the document]
Comment (hav-noms) [reply SIPS/sip-30.md]: fixed

File: SIPS/sip-30.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+- A new Depot will be required which will go back to using the original ERC20 approve, transferFrom workflow. This could be an opportunity to makes some additional improvements to the Depot such as making it upgradable. Putting it behind a proxy and giving it an external state contract so its logic can be upgraded.
+
+## Rationale
+
+<!--The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+
+- Removing ERC223 will no longer show the transfer errors in contract to contract transfers. [etherscan](https://etherscan.io/address/0xe9cf7887b93150d4f2da7dfc6d502b216438f244)
+- This will also save 200K gas per contract to contract transfer. [github code reference](https://github.com/Synthetixio/synthetix/blob/master/contracts/TokenFallbackCaller.sol#L52)
```
Comment (justinjmoses) [inline SIPS/sip-30.md]: [np] Use the actual version in the link so it works in perpetuity: https://github.com/Synthetixio/synthetix/blob/v2.14.0/contracts/TokenFallbackCaller.sol#L52

File: SIPS/sip-30.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+- The current [Depot](https://contracts.synthetix.io/Depot) will no longer be able to accept sUSD deposits effectivly putting it to its end of life.
+- A new Depot will be required which will go back to using the original ERC20 approve, transferFrom workflow. This could be an opportunity to makes some additional improvements to the Depot such as making it upgradable. Putting it behind a proxy and giving it an external state contract so its logic can be upgraded.
+
+## Rationale
+
+<!--The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+
+- Removing ERC223 will no longer show the transfer errors in contract to contract transfers. [etherscan](https://etherscan.io/address/0xe9cf7887b93150d4f2da7dfc6d502b216438f244)
```
Comment (justinjmoses) [inline SIPS/sip-30.md]: [np] instead of `etherscan` can you say Uniswap sETH exchange?

File: SIPS/sip-30.md (hunk 2 above)
Comment (hav-noms) [reply SIPS/sip-30.md]: done

File: SIPS/sip-30.md (hunk 3 above)
Comment (hav-noms) [reply SIPS/sip-30.md]: much better. done

File: SIPS/sip-30.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+## Rationale
+
+<!--The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+
+- Removing ERC223 will no longer show the transfer errors in contract to contract transfers. [Uniswap sETH exchange](https://etherscan.io/address/0xe9cf7887b93150d4f2da7dfc6d502b216438f244)
+- This will also save 200K gas per contract to contract transfer. [github code reference](https://github.com/Synthetixio/synthetix/blob/v2.14.0/contracts/TokenFallbackCaller.sol#L52)
+- Reclaim byte code space for SNX contract deployment by removeing the ERC223 implementation in `ExternStateToken.sol`.
```
Comment (justinjmoses) [inline SIPS/sip-30.md]: [np] removeing => removing

Comment (hav-noms) [reply SIPS/sip-30.md]: done

**Pull request merged**